src/wrapper/signature.rs

```diff
@@ -77,10 +77,9 @@ pub struct TypeSignature {
 }
 
 impl TypeSignature {
-    /// Parse a signature string into a TypeSignature enum.
-    ///
     /// Clippy suggests implementing `FromStr` or renaming it which is not possible in our case.
     #[allow(clippy::should_implement_trait)]
+    /// Parse a signature string into a TypeSignature enum.
```

Comment:
Such placement doesn't look natural ­— can't the documentation go first (as the most important part) — and then annotations with a comment?

Reply:
Are you talking about something like this?
```
    /// Parse a signature string into a TypeSignature enum.
    #[allow(clippy::should_implement_trait)]
    /// Clippy suggests implementing `FromStr` or renaming it which is not possible in our case.
    fn from_str()
```

Reply:
Done.